src/internal_modules/roc_node/context.cpp

```diff
@@ -15,10 +15,11 @@ namespace node {
 
 Context::Context(const ContextConfig& config, core::IArena& arena)
     : arena_(arena)
-    , packet_factory_(arena_)
-    , byte_buffer_factory_(arena_, config.max_packet_size)
-    , sample_buffer_factory_(arena_, config.max_frame_size / sizeof(audio::sample_t))
+    , packet_pool_("packet_pool", arena_)
+    , byte_buffer_pool_("byte_buf_pool", arena_, sizeof(core::Buffer<uint8_t >) + sizeof(uint8_t) * config.max_packet_size)
+    , sample_buffer_pool_("sample_buf_pool", arena_, sizeof(core::Buffer<audio::sample_t >) + sizeof(audio::sample_t) * config.max_frame_size)
     , encoding_map_(arena_)
+    // TODO:
```

Comment:
@gavv It seems the network_loop_ below requires the factories. Do I pass the pools into it and let it create it's own factories?

Reply:
Yep, sounds good. roc_node will instantiate arenas pools and pass them to roc_netio and roc_pipeline. They, in turn, will instantiate factories when needed.